pkg/controller/testutil/test_utils.go

```diff
@@ -417,6 +422,26 @@ func (f *FakeRecorder) AnnotatedEventf(obj runtime.Object, annotations map[strin
 	f.Eventf(obj, eventtype, reason, messageFmt, args...)
 }
 
+func (f *FakeRecorder) AssertEventReasons(events []v1.Event) bool {
+	f.Lock()
+	defer f.Unlock()
+	return f.assertReasons(events, f.Events)
+}
+
+func (f *FakeRecorder) assertReasons(test []v1.Event, expect []*v1.Event) bool {
+	if len(test) != len(expect) {
+		f.T.Errorf("AssertList: event lists are different lengths")
+		return false
+	}
+	for i := range test {
+		if expect[i].Reason != test[i].Reason {
+			f.T.Errorf("AssertList: expected %#v, got %#v", expect[i].Reason, test[i].Reason,)
+			return false
+		}
+	}
+	return true
+}
```

Comment:
Can you try to use [slices.EqualFunc](https://pkg.go.dev/golang.org/x/exp/slices#EqualFunc)

Reply:
done